nym-vpn-desktop/src-tauri/src/commands/db.rs

```diff
@@ -0,0 +1,138 @@
+    debug!("db_get_batch");
+    let err_msg = "Failed to retrieve db data".to_string();
+    let entry_node_location = db
+        .get_typed::<NodeLocation>(Key::EntryNodeLocation)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let exit_node_location = db
+        .get_typed::<NodeLocation>(Key::ExitNodeLocation)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let vpn_mode = db
+        .get_typed::<VpnMode>(Key::VpnMode)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let ui_theme = db
+        .get_typed::<UiTheme>(Key::UiTheme)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let entry_location_enabled = db
+        .get_typed::<bool>(Key::EntryLocationEnabled)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let ui_root_font_size = db
+        .get_typed::<u32>(Key::UiRootFontSize)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let monitoring = db
+        .get_typed::<bool>(Key::Monitoring)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let autoconnect = db
+        .get_typed::<bool>(Key::Autoconnect)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+
+    trace!(
+        r"
+entry_node_location: {:?}
+exit_node_location: {:?}
+vpn_mode: {:?}
+ui_theme: {:?}
+entry_location_enabled: {:?}
+ui_root_font_size: {:?}
+monitoring: {:?}
+autoconnect: {:?}",
+        entry_node_location,
+        exit_node_location,
+        vpn_mode,
+        ui_theme,
+        entry_location_enabled,
+        ui_root_font_size,
+        monitoring,
+        autoconnect
+    );
+
+    Ok(AppData {
+        entry_node_location,
+        exit_node_location,
+        vpn_mode,
+        ui_theme,
+        entry_location_enabled,
+        ui_root_font_size,
+        monitoring,
+        autoconnect,
+    })
+}
+
+#[instrument(skip(db))]
+#[tauri::command]
+pub async fn db_get(db: State<'_, Db>, key: Key) -> Result<Option<JsonValue>, CmdError> {
+    debug!("db_get");
+    db.get(key).map_err(|_| {
+        CmdError::new(
+            CmdErrorSource::InternalError,
+            format!("Failed to get key [{key}]"),
+        )
+    })
+}
+
+#[instrument(skip(db))]
+#[tauri::command]
+pub async fn db_set(
+    db: State<'_, Db>,
+    key: Key,
+    value: JsonValue,
+) -> Result<Option<JsonValue>, CmdError> {
+    debug!("db_set");
+    db.insert(key, &value).map_err(|e| match e {
+        DbError::Serialize(e) => CmdError::new(
+            CmdErrorSource::CallerError,
+            format!("Failed to insert key, bad JSON input: {e}"),
+        ),
+        _ => CmdError::new(
+            CmdErrorSource::InternalError,
+            format!("Failed to insert key: {e}"),
+        ),
+    })
```

Comment:
does it make sense to expose raw database operations to the frontend? shouldn't we just have helpers like `set_autoconnect` instead?

Reply:
> does it make sense to expose raw database operations to the frontend

In short: Absolutely **YES** and this is the main reason of that new implem

Long: this is already explained in PR description, but I'll try to explain it using a different wording
With storing data within a simple file (as it is currently)
you have to provide a dedicated tauri API for each value frontend want to get/set
That means you want to set/get the uiTheme from the frontend and saveit, you have to code 2tauri dedicated "commands"
one to set, one to get
you have to repeat that for any other saved value, so for selected entry location, exit location and so on.
It's a bad design, it creates code duplication, exact same logic is repeated you get it.